src/utils/camera.ts

```diff
@@ -0,0 +1,60 @@
+import type { SiteType } from '@/components/Live/useDataSitesLive';
+import type { CameraType } from '@/services/camera';
+import type { CameraInfosFromPi } from '@/services/live';
+
+export interface CameraFullInfosType extends CameraType {
+  ip?: string;
+  type?: string;
+  azimuths?: number[];
+  poses?: number[];
+}
+
+export const getSiteByCameraName = (
+  sites: SiteType[],
+  cameraName: string
+): SiteType | null => {
+  return (sites.find((site) =>
+    site.cameras.map((camera) => camera.name).includes(cameraName)
+  ) ?? sites.length > 0)
+    ? sites[0]
+    : null;
+};
```

Comment:
this probably does not work as expected: if the list is non-empty, the function returns the first site (and the cameraName isn't used). I'm guessing you wanted to do something like:
```suggestion
export const getSiteByCameraName = (
  sites: SiteType[],
  cameraName: string
): SiteType | null => {
    return sites.find((site) =>
      site.cameras.some((camera) => camera.name === cameraName)
    ) ?? (sites.length > 0 ? sites[0] : null);
};
```

But do we event want to return `sites[0]` when no camera with the right name is found?

```suggestion
export const getSiteByCameraName = (
  sites: SiteType[],
  cameraName: string
): SiteType | null => {
    return sites.find((site) =>
      site.cameras.some((camera) => camera.name === cameraName)
    ) ?? null;
};
```

Reply:
Yep, there is a problem with my (), fixed. I left only "?? null"